pkg/config/legacy: add missing trace agent properties to legacy config converter.

gbbr opened the pull request:
Adds missing trace agent properties to legacy config converter. As a follow-up to #2678

gbbr committed:
pkg/config/legacy: add tests and extra trace agent sections

gbbr committed:
pkg/config/legacy: fix proxy converter to match config keys

gbbr committed:
pkg/config/legacy: add missing trace agent keys and tests

gbbr commented:
@olivielpeau can you PTAL if this is more along the lines of what you meant in the other PR? If yes, I will test this PR directly in trace and potentially merge it if afterwards if all is well with you too and the team.

gbbr committed:
pkg/config/legacy: add converter coverage for all trace keys

olivielpeau commented on pkg/config/legacy/converter.go:
```diff
@@ -132,44 +137,120 @@ func FromAgentConfig(agentConfig Config) error {
 	if agentConfig["apm_enabled"] != "" {
-		if enabled, err := isAffirmative(agentConfig["apm_enabled"]); err == nil && !enabled {
+		if enabled, err := isAffirmative(agentConfig["apm_enabled"]); err == nil {
 			// apm is enabled by default, convert the config only if it was disabled
```
this comment is outdated

olivielpeau commented on pkg/config/legacy/converter.go:
```diff
@@ -132,44 +137,120 @@ func FromAgentConfig(agentConfig Config) error {
 	if agentConfig["non_local_traffic"] != "" {
-		if enabled, err := isAffirmative(agentConfig["non_local_traffic"]); err == nil && enabled {
+		if enabled, err := isAffirmative(agentConfig["non_local_traffic"]); err == nil {
 			// trace-agent listen locally by default, convert the config only if configured to listen to more
```
outdated comment

olivielpeau commented on pkg/config/legacy/converter.go:
```diff
@@ -132,44 +137,120 @@ func FromAgentConfig(agentConfig Config) error {
 	}
 
-	if sampleRate, err := strconv.ParseFloat(agentConfig["extra_sample_rate"], 64); err == nil {
+	if sampleRate, err := strconv.ParseFloat(agentConfig["trace.sampler.extra_sample_rate"], 64); err == nil {
```
why are these trace options handled both here and in `extractTraceAgentConfig`?

gbbr replied:
Accident, there were a couple of duplicates. I've moved everything trace related into `extractTraceAgentConfig` and removed duplicates.

olivielpeau reviewed (dismissed):
Left a few comments, overall LGTM 👍

I'm interested in how you'll be using this in the trace-agent, please ping me when those are ready :)

gbbr committed:
address PR comments

gbbr commented:
Thanks for looking. I'm going to try it out in trace agent and see if anything is missing. Will ping you on both PRs.

gbbr committed:
set rates as map to allow keys containing periods ('.')

gbbr commented:
@olivielpeau PTAL. Addressed your comments and submitted another minor fix. Also tried it out on trace agent. Will open PR soon once this gets merged (so I can vendor the commit inside trace-agent).

gbbr committed:
fix tests

olivielpeau reviewed (approved):
LGTM! 👍 

I'll let you get approval from the Agent guild as well ;)

Had a cursory look at the trace-agent PR as well and it also looks good to me

mfpierre reviewed (approved):
LGTM 👍
